components/users/rest/src/main/java/org/phenotips/users/rest/UsersResource.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.users.rest;
+
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+
+/**
+ * Root resource for working with users and groups of users.
+ *
+ * @version $Id$
+ * @since 1.4
+ */
+@Path("/users")
```

Comment:
I don't like the API.
* First, `users` is inappropriate. According to @sdumitriu , the correct term that captures both users and groups is `principals`.
* Second, it's inconsistent with our other services and it doesn't follow best practices. A better path is `principals/suggest`. Moreover, the root `/principals` should list all users and groups, and `/principals/{id}` should provide information about that particular principal, but this is out of the scope of this task for now.
* Both groups and users should be suggested by default.